packages/shared/src/features/onboarding/steps/FunnelReadingReminder.tsx

```diff
@@ -1,21 +1,46 @@
 import type { ReactElement } from 'react';
-import React from 'react';
+import React, { useCallback, useEffect, useRef } from 'react';
 import type { FunnelStepReadingReminder } from '../types/funnel';
 import { FunnelStepTransitionType } from '../types/funnel';
 import { ReadingReminder } from '../../../components/onboarding';
 import { withIsActiveGuard } from '../shared/withActiveGuard';
+import { useViewSize, ViewSize } from '../../../hooks';
+import { usePushNotificationContext } from '../../../contexts/PushNotificationContext';
 
 function FunnelReadingReminderComponent({
   parameters: { headline },
   onTransition,
 }: FunnelStepReadingReminder): ReactElement | null {
-  const handleSubmit = ({ skipped }: { skipped?: boolean } = {}) => {
-    onTransition({
-      type: skipped
-        ? FunnelStepTransitionType.Skip
-        : FunnelStepTransitionType.Complete,
-    });
-  };
+  const isMobile = useViewSize(ViewSize.MobileXL);
+  const { isPushSupported, isInitialized } = usePushNotificationContext();
+  const haveEvaluated = useRef(false);
+
+  const handleSubmit = useCallback(
+    ({ skipped }: { skipped?: boolean } = {}) => {
+      onTransition({
+        type: skipped
+          ? FunnelStepTransitionType.Skip
+          : FunnelStepTransitionType.Complete,
+      });
+    },
+    [onTransition],
+  );
+
+  useEffect(() => {
+    if (haveEvaluated.current) {
```

Comment:
Should we check here first once it is initialized? So that if not, we will return early?

Reply:
Ah, then it makes sense. Yeah was thinking about that maybe does not get to true on certain scenarios. Kindly ignore.